compare/compare.c

```diff
@@ -18,9 +18,9 @@
 
 #include "image.h"
 #include "../libass/ass.h"
+#include "../libass/ass_filesystem.h"
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <dirent.h>
 #include <string.h>
```

Comment:
Using libass internal functions means this can no longer be built with dynamic libass

Either avoid this by just not supporting `compare` with MSVC, or force static linkage like for `checkasm` i both autotools and meson

Reply:
> by just not supporting `compare` with MSVC

Don’t we need it for tests? Isn’t this (it being unsupported with MSVC) the current status quo that this PR is aiming to change?

Reply:
We need it for full test coverage, yes. If the goal of the PR is to get the Windows threading implementation tested in CI, MinGW is sufficient if i’m not mistaken about it also using Win32-native API. If the goal is to get specifically MSVC tested in CI, then yeah, omitting `compare` support goes against that

Reply:
Not sure about threads, but at the very least, MSVC builds use different atomics from MinGW.